src/main/java/com/iota/iri/cache/CacheManager.java

```diff
@@ -0,0 +1,39 @@
+package com.iota.iri.cache;
+
+import java.util.Collection;
+
+/**
+ * Cache Manager
+ */
+public interface CacheManager {
+
+    /**
+     * Get the names of all caches
+     * 
+     * @return A collection of cache names
+     */
+    Collection<String> getCacheNames();
+
+    /**
+     * Get the cache with the specified name
+     * 
+     * @param name The name of the cache to get
+     * @return The cache with the specified name
+     */
+    Cache getCache(String name);
```

Comment:
Maybe instead of looking by name look by Class.
I don't really think you need names for the classes.

You can do something generic like:
 `<T extends Persistable> Cache getCache(Class<T> type)`

This kust makes it clear what kind of objects you can cache. I just less like using strings.